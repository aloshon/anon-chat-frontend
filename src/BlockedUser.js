import {useSelector, useDispatch} from "react-redux";
import {unblockUser} from "./Actions/actionCreators";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import "./BlockedUser.css";

/** 
 * BlockedUser renders the username of those who are on the block list
 */
const BlockedUser = ({blockedUser}) => {
    const dispatch = useDispatch();
    const darkMode = useSelector(state => state.darkMode);
    return (
        <Card className="blocked-contact-card" style={{backgroundColor: darkMode.card}}>
            <Card.Body>
                <Card.Title>{blockedUser}</Card.Title>
                <Button data-testid="unblock"
                variant="success"
                onClick={() => dispatch(unblockUser(blockedUser))}>
                    Unblock
                </Button>
            </Card.Body>
        </Card>
    )
}

export default BlockedUser;