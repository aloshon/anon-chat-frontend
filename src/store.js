import  { composeWithDevTools } from "redux-devtools-extension";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import autoMergeLevel2 from "redux-persist/lib/stateReconciler/autoMergeLevel2";

import thunk from "redux-thunk";
import rootReducer from "./Reducers/rootReducer";
import { createStore, applyMiddleware } from "redux";

// only keep the darkMode reducer persistent in local storage
const persistConfig = {
  key: "root",
  whitelist:['darkMode'],
  storage,
  stateReconciler: autoMergeLevel2
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  composeWithDevTools(
    applyMiddleware(thunk),
  )
);

export const persistedStore = persistStore(store);