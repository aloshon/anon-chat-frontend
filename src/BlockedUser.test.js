import { render, screen } from '@testing-library/react';
import BlockedUser from './BlockedUser';
import { Provider } from "react-redux";
import { store } from "./store"
import {BrowserRouter} from 'react-router-dom';


describe("BlockedUser should render blocked user's username and button to unblock", () => {
    const blockedUser = "user1";

    const {asFragment} = render(
      <Provider store={store}>
        <BrowserRouter>
            <BlockedUser blockedUser={blockedUser}/>
        </BrowserRouter>
      </Provider>
    );
    test("renders group chat card with basic info", () => {
      const username = screen.getByText("user1");
      expect(username).toBeInTheDocument();
      const unblock = screen.getByTestId("unblock");
      expect(unblock).toBeInTheDocument();
    });
  
    test("matches snapshot", () => {
      expect(asFragment()).toMatchSnapshot();
    });
    
  });