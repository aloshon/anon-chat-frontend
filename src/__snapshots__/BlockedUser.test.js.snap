// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockedUser should render blocked user's username and button to unblock matches snapshot 1`] = `<DocumentFragment />`;
